### PR TITLE
add integration test section to the SSO mission (except node)

### LINKED ISSUE #721: Secured mission: Add documentation for running integration tests
The secured Booster docs requires a refactor of the integration tests section.
Related: Issue #711  
EDIT based on conversation with Heiko: 
The tests moved to the booster itself, whereas previously they did reside in the sso module.
Apart from that nothing should have changed (i.e. the instruction to the run the tests remain the same)

### PR DESCRIPTION
**Pull request by inoxx03**
fixes #721

### CODE REVIEW

File: docs/topics/secured-mission-integration-tests.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+$ mvn clean verify -Popenshift,openshift-it -DSSO_AUTH_SERVER_URL=$(oc get route secure-sso -o jsonpath='https://{.spec.host}/auth')
+--
++
+[WARNING]
+--
+Ensure that you add view access rights for the service account before executing tests.
+
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ oc policy add-role-to-user view -n $(oc project -q) -z default
```
Comment (Ladicek) [inline docs/topics/secured-mission-integration-tests.adoc]: Not sure if this is really needed.

**Comment by Ladicek** [comment]
Looks good to me and I know this works fine with Swarm, but I _don't know_ if this works with Vert.x and Spring Boot. CC @cescoffier / @spisiakm for Vert.x and @alesj for Spring Boot.

**Comment by rhoads-zach** [comment]
LGTM, merging.

File: docs/topics/secured-mission-integration-tests.adoc (hunk 1 above)
Comment (inoxx03) [reply docs/topics/secured-mission-integration-tests.adoc]: I've been wondering about that one, too. I see that the callout about view access is included for configmap. I don't believe that it is necessary for the Secured boosters, though. I'll remove it

**Comment by inoxx03** [comment]
@cescoffier @spisiakm Would these steps work for Vert.x, or is there anything I missed?
@alesj Would the same steps work for Spring Boot, too?

File: docs/topics/secured-mission-integration-tests.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+[source,bash,options="nowrap",subs="attributes+"]
+--
+oc apply -f service.sso.yaml
+--
++
+. Execute the following command to run the tests. Use the following syntax to pass in the URL of the {RHSSO} server as a parameter:
++
+[source,bash,options="nowrap",subs="attributes+"]
+--
+$ mvn clean verify -Popenshift,openshift-it -DSSO_AUTH_SERVER_URL=$(oc get route secure-sso -o jsonpath='https://{.spec.host}/auth')
```
[the remark this answers is not in the document]
Comment (inoxx03) [reply docs/topics/secured-mission-integration-tests.adoc]: Done

**Comment by tradej** [comment]
Please, do not merge, I need to review the updated version first.

File: docs/topics/secured-mission-integration-tests.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ladicek) [reply docs/topics/secured-mission-integration-tests.adoc]: @tradej I wonder why you chose this over `-o jsonpath='https://{.spec.host}/auth'`, which I'm gonna argue is a lot easier to read. Also when passing the URL to the test, I'm gonna argue that the final `\n` shouldn't really be there (but I guess it's stripped away automatically either by the shell or by the JVM).

**Comment by tradej** [comment]
Good job; I have filed a PR with some changes at inoxx03/documentation#4.

**Comment by inoxx03** [comment]
@tradej Thank you for the review PR, I've incorporated your fixes into this PR:)

**Comment by rhoads-zach** [comment]
merging for now, will accept follow PRs with feedback from @cmoulliard and spring boot.

**Pull request merged**